python/lsst/multiprofit/fitting/fit_psf.py

```diff
@@ -0,0 +1,639 @@
+        for flag in (col for col in params.keys() if col.endswith("_flag")):
+            if params[flag]:
+                raise PsfRebuildFitFlagError(f"Failed to rebuild PSF; {flag} set")
+
+        for name, param in self.parameters.items():
+            param.value = params[f"{self.config.prefix_column}{name}"]
+
+    def model_post_init(self, __context):
+        self.config.freeze()
+        n_component_configs = len(self.component_configs)
+        n_components = len(self.psf_model.components)
+        if n_components != n_component_configs:
+            raise AssertionError(f"{n_components=} != {n_component_configs=}")
+
+    @cached_property
+    def parameters(self) -> dict[str, g2f.ParameterD]:
+        parameters = {}
+        has_prefix_group = self.config.model.has_prefix_group()
+        components = self.psf_model.components
+        idx_comp_first = 0
+        for name_group, config_group in self.componentgroupconfigs.items():
+            prefix_group = f"{name_group}_" if has_prefix_group else ""
+            is_fractional = config_group.is_fractional
+            multicen = len(config_group.centroids) > 1
+            configs_comp = config_group.get_component_configs()
+            idx_last = len(configs_comp) - 1
+            n_params_flux_frac = 0
+
+            for idx_comp_group, (name_comp, config_comp) in enumerate(configs_comp.items()):
+                is_last = idx_comp_group == idx_last
+                component = components[idx_comp_first + idx_comp_group]
+                label_size = config_comp.get_size_label()
+                prefix_comp = f"{prefix_group}{name_comp}{'_' if name_comp else ''}"
+                if multicen or (idx_comp_group == 0):
+                    prefix_cen = prefix_comp if multicen else prefix_group
+                    parameters[f"{prefix_cen}cen_x"] = component.centroid.x_param
+                    parameters[f"{prefix_cen}cen_y"] = component.centroid.y_param
+                if not config_comp.size_x.fixed:
+                    parameters[f"{prefix_comp}{label_size}_x"] = component.ellipse.size_x_param
+                if not config_comp.size_y.fixed:
+                    parameters[f"{prefix_comp}{label_size}_y"] = component.ellipse.size_y_param
+                if not config_comp.rho.fixed:
+                    parameters[f"{prefix_comp}rho"] = component.ellipse.rho_param
+
+                # TODO: return this to component.integralmodel
+                # when binding for g2f.FractionalIntegralModel is fixed
+                params_flux = get_params_uniq(component, fixed=False, nonlinear=False)
+                has_params_flux = not config_comp.flux.fixed and (
+                    (not is_fractional) or (idx_comp_group == 0)
+                )
+                if len(params_flux) != has_params_flux:
+                    raise RuntimeError(
+                        f"{params_flux=} has len={len(params_flux)} but expected {has_params_flux}"
+                    )
```

Comment:
This seems weird to me. You're checking the number of unique parameters (`len(params_flux`) with a boolean (`has_params_flux`). I'm guessing that the idea is that when has_params_flux is false, then there should be no unique params, and when has_params_flux is true there should be one, and only one param, right?

I get that python allows for this type of comparison, but it's a bit confusing and should either come with a comment or be more explicit to make it clear what it's doing.

```suggestion
                params_flux = get_params_uniq(component, fixed=False, nonlinear=False)
                has_params_flux = not config_comp.flux.fixed and (
                    (not is_fractional) or (idx_comp_group == 0)
                )
                if has_params_flux and n_params != 1 or not has_params_flux and n_params != 0:
                    raise RuntimeError(
                        f"{params_flux=} has len={len(params_flux)} but expected {has_params_flux}"
                    )
```

Reply:
I clarified it in a somewhat different manner.